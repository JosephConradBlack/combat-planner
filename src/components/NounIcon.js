import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Image } from 'semantic-ui-react';

const StyledImage = styled(Image)`
  width: 33.03px;
  height: 28px;
  display: inline-block;
  margin-right: 3.5px;
`;

const NounIcon = props => <StyledImage {...props} />;

NounIcon.propTypes = {
  src: PropTypes.string.isRequired
};
export default NounIcon;
