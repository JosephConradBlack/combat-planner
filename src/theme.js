export default {
  h1: {
    fontSize: '6em',
    fontWeight: 300,
    letterSpacing: '0.06em',
    subheader: {
      fontSize: '1em',
      fontWeight: 'lighter'
    }
  }
};
