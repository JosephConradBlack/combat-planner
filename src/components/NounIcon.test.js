import React from 'react';
import { ThemeProvider } from 'styled-components';

import theme from '../theme';
import NounIcon from './NounIcon';

import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('NounIcon', () => {
  it('renders with src', () => {
    const component = shallow(
      <ThemeProvider theme={theme}>
        <NounIcon src="icon_sword.svg" />
      </ThemeProvider>
    ).dive();

    expect(toJson(component)).toMatchSnapshot();
  });
});
