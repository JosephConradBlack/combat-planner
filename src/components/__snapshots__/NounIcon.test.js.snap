// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NounIcon renders with src 1`] = `
<Styled(Image)
  src="icon_sword.svg"
/>
`;
